### DISCUSSION TITLE
Help with Reverse Proxy Configuration for Multiple Websites with Nextcloud AIO

### DISCUSSION THREAD

**Discussion by B1N4RY-P4R45173** [nextcloud/all-in-one#5415] [Questions] [answered]
Hello everyone,

I’m running **Nextcloud AIO** with the standard Docker setup. Currently, everything works well with a single domain, but I now want to **host an additional website** on the same server.

## My Goal
Requests to **cloud.example.com** should be forwarded to the **Nextcloud Apache container** (serving the Nextcloud UI/Login page).  
   - **HTTPS should be enabled** for the Nextcloud website (cloud.example.com) through the reverse proxy.
2. Requests to **example.com** should be directed to a **separate web server**.  

## The Problem
- The **Nextcloud AIO master container** is already listening on **port 80** (used for ACME challenges to obtain certificates).
- I planned to set up **Nginx as a reverse proxy** (though I am open for any other better alternatives) to handle traffic for both domains, but I’m stuck because both Nginx and the master container need to bind to **port 80**.

## What I Need
- The ACME challenges for Nextcloud AIO to continue working smoothly.
- The reverse proxy to handle incoming requests to the two domains and route them to the appropriate services.
- If possible, I’d prefer a solution that doesn’t rely on custom port numbers for the websites.
- Also I would be using **HTTPS** for my website (example.com). (just adding this point so that I don't face any issues while configuring it)

## Setup

My server is running windows 11 with aio version (9.6.0).

I am using the following docker run command
```
docker run ^
--init ^
--sig-proxy=false ^
--name nextcloud-aio-mastercontainer ^
--restart always ^
--publish 80:80 ^
--publish 8080:8080 ^
--publish 8443:8443 ^
--env APACHE_PORT=11000 ^
--env APACHE_IP_BINDING=0.0.0.0 ^
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config ^
--volume //var/run/docker.sock:/var/run/docker.sock:ro ^
--env NEXTCLOUD_DATADIR="/run/desktop/mnt/host/d/ncdata" ^
--env NEXTCLOUD_MOUNT="/run/desktop/mnt/host/d/" ^
nextcloud/all-in-one:latest
```
Also I wanted to add that now if I visit https://cloud.example.com:11000 I get `ERR_SSL_PROTOCOL_ERROR` I think this was explained in [here](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#5-optional-get-a-valid-certificate-for-the-aio-interface) but I stuck at the Caddyfile configurations

I have gone through [Reverse Proxy Guide](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md) and Discussion #588  but I am still lost and confused.

I’m fairly new to these configurations, so it's possible I missed something or lack the knowledge to set it up correctly. If there’s any simpler way to achieve this, or if you have other suggestions, please let me know.

Thanks in advance!

**Comment by szaimen** [comment, marked as answer]
Hi, can you follow https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#6-how-to-debug-things?

**Comment by B1N4RY-P4R45173** [comment]
Thanks @szaimen for the quick reply. I have gone through that but I am failing somewhere. Would appreciate your help

**Reply by szaimen** [reply]
You might want to have a look at https://github.com/nextcloud/all-in-one/discussions/575 and ask there if you should run into issues
